subscription-sender/internal/repo/subsctiption.go

```diff
@@ -0,0 +1,67 @@
+package repo
+
+import (
+	"context"
+	"github.com/lib/pq"
+	"time"
+
+	"github.com/RinaDish/subscription-sender/internal/services"
+)
+
+type message struct {
+	ID int `gorm:"id"`
+	Rate float64 `gorm:"rate"`
+	Emails pq.StringArray `gorm:"type:text[]"`
+	CreatedAt time.Time `gorm:"created_at"`
+	EventID uint8 `gorm:"event_id"`
+	EventType string `gorm:"event_type"`
+	SendingTime time.Time `gorm:"sending_time"`
+	Sent bool `gorm:"sent"`
+}
+
+func (repo *Repository) SetMessages(ctx context.Context, m services.Message) error {
+	msg := &message{
+		ID: m.ID,
+		Rate: m.Rate,
+		Emails: pq.StringArray(m.Emails),
+		CreatedAt: m.CreatedAt,
+		EventID: m.EventID,
+		EventType: m.EventType,
+		SendingTime: m.SendingTime,
+		Sent: m.Sent,
+	}
+	return repo.DB.Table("messages").Create(&msg).WithContext(ctx).Error
+}
+
+func (repo *Repository) GetMessages(ctx context.Context) ([]services.Message, error) {
+
+	msgsList := make([]message, 0)
+
+	currentTime := time.Now()
+
+	err := repo.DB.Table("messages").WithContext(ctx).Where("sent = ?", false).Where("sending_time < ?", currentTime).Find(&msgsList).Error
```

Comment:
Oh, seems like not clear contract. So when i'm calling `GetMessages` I'm expecting to actually get messages, not with some additional logic. So or rename method, or make filtration on client side who's calling this method

Reply:
Renamed